src/webview/chat_js.ts

```diff
@@ -1,5 +1,5 @@
-import * as vscode from "vscode";
 export const chatJs = () => `
+const vscode = acquireVsCodeApi();
```

Comment:
This was the root cause of the issue. The call to this method was within the sendMessage function, therefore the extension was unaware of the events.